docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying       5         3         3            3         
+```
+
+Failure:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Stuck           5         4         4            3 
+```
+
+Success:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Active          5         5         5            5         
+```
+
+Updating an environment
+-----------------------
+
+Typically, you'd update an environment in order to:
+
+-   update the task definition that should be running
+-   change the set of instances on which to run tasks
+-   change the deployment configuration (i.e. to change health
+    thresholds) (TODO: Maybe require new env for this?)
+
+### Updating the task definition
+
+If you don't specify all attributes, you must specify the revision you
+wish to base your changes on. This prevents potential conflicts if
+multiple callers attempt to update an environment at the same time.
+
+``` {.shell}
+$ aws ecs update-environment --environment SomeEnvironment/Prod --revision 1 --task-definition "log-daemon:2"
+{ "EnvironmentRevision": 1 }
+```
+
+At this point, nothing has actually happened to your cluster. You can
+see this by showing your environment's status:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Active     5         5         5            5        
+SomeEnvironment/Prod:2    Created    0         0         0            0        
+```
+
+In order to make this configuration Active, you first have to Deploy the
+revision:
+
+``` {.shell}
+$ aws ecs deploy-environment --environment-name SomeEnvironment/Prod --environment-revision 2
+{ "DeploymentId": "SomeEnvironment/Prod:2@2017090611472301" }
+```
+
+This will mark the revision as Deploying, and start launching tasks to
+meet the deployment configuration's constraints. In the case of a Daemon
+environment, the desired target is to have one copy of the task for
+every instance in the target Instance Group:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Undeploying  0         5         5            5        
+SomeEnvironment/Prod:2    Deploying    5         0         0            0        
+```
+
+Once the deployment reaches a stead state, the old revision is marked
```

Comment:
nit: stead**y** state

Reply:
Thanks; there's a bunch of typos I'll fix shortly. Any other feedback would be appreciated as well.